File: xhs_growth/tools/xhs/analytics.py
"""XHS analytics tools."""

from langchain_core.tools import tool


@tool
async def analytics_reader(post_id: str) -> dict:
    """读取小红书帖子数据分析"""
    return {
        "post_id": post_id,
        "views": 0,
        "likes": 0,
        "collects": 0,
        "comments": 0,
        "shares": 0,
        "engagement_rate": 0.0,
    }


@tool
async def pattern_detector(time_range: str = "7d") -> list[dict]:
    """检测内容表现模式 — 识别哪些类型/时段/标签表现最好"""
    return [{"pattern": "示例模式", "confidence": 0.0, "time_range": time_range}]


@tool
async def report_generator(account_id: str, period: str = "weekly") -> str:
    """生成增长报告"""
    return f"增长报告: account={account_id}, period={period}"

File: frontend/src/types/analytics.ts
// 增长报告
export interface GrowthReport {
  account_id: string
  period: 'daily' | 'weekly' | 'monthly'
  report?: string
  metrics?: {
    total_posts: number
    total_engagement: number
    avg_engagement_rate: number
    best_post_title: string
    trend_topics: string[]
  }
  insights?: Array<{
    type: 'trend' | 'opportunity' | 'warning' | 'info'
    message: string
  }>
  generated_at?: string
}

// 帖子表现
export interface PostPerformance {
  /** Workflow publish id or imported creator-center note id. */
  id?: string
  title: string
  likes: number
  comments: number
  collects: number
  shares: number
  views: number
  engagement_rate: number
  published_at: string
  /** Source of the row; imported rows are eligible for note-quality drill-down. */
  source?: string
  account_id?: string
  workflow_thread_id?: string
  platform_post_id?: string
  link_status?: 'linked' | 'unmatched' | 'ambiguous' | string
  linked_note_id?: string
}

// 性能数据
export interface PerformanceData {
  account_id: string
  posts: PostPerformance[]
  /** Number of all posts in the selected period, independent of page limit. */
  total?: number
}

export interface AnalyticsPeriodMetrics {
  posts: number
  views: number
  likes: number
  comments: number
  collects: number
  shares: number
  engagement: number
  avg_engagement_rate: number
}

export interface AnalyticsPeriodSummary {
  period: 'daily' | 'weekly' | 'monthly'
  current: AnalyticsPeriodMetrics
  previous: AnalyticsPeriodMetrics
}

// 成本数据
export interface CostData {
  total_cost_usd: number
  today_cost_usd: number
  budget_remaining_usd?: number
  by_model?: Record<string, number>
  circuit_open: boolean
}
